src/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import injectSheetDefault from 'react-jss'
+import { pure } from 'recompose'
+
+
+type StyledElementType = Function & { Tag: string, styles: Object }
+
+
+export const prepareStyled = (injectSheet: Function = injectSheetDefault) =>
+  (
+    Element: string | StyledElementType,
+    styles: Object,
+    baseStyles: ?Object = {},
+  ): StyledElementType => {
+    const {
+      Tag,
+      styles: inheritStyles = {},
+    }: {
+      Tag: string,
+      styles?: Object
+    } = typeof Element === 'string' ? { Tag: Element } : Element
+
+    const elementStyles = { ...inheritStyles, ...styles }
+
+    const StyledElement = pure(
```

Comment:
not sure shallowEqual should be implemented by default, does styled-components has it ?